Automattic/woocommerce-payments
# Do not set UPE billing details when changing a subscription payment method

Fixes #2917

#### Changes proposed in this Pull Request

Tweaks a condition in `upe.js` to ensure the `billingDetails` UPE config is not set when changing payment methods for an existing subscription.

Prior to this change, attempting to change the payment methods for an existing subscription would result in the following error when entering new card details.

<img width="829" alt="Screen Shot 2021-09-15 at 3 43 32 pm" src="https://user-images.githubusercontent.com/1527164/134107492-05952776-f1fe-45a0-b930-7a5943f27c60.png">

#### Testing instructions

* As a Merchant, ensure UPE checkout is enabled and the WooCommerce Subscriptions plugin is installed. 
* As a Shopper, purchase any subscription product.
* Navigate to **My account → Subscriptions** and open the newly created subscription entry.
* Observe 3 buttons for “Actions”. Click “Change payment” button.
* In the second block, select “Use a new payment method” option.
* Provide `4242424242424242` as the test card number, future expiration date, and a CVC code.
* Click the “Change payment method” button.
* Ensure the payment method is updated successfully.

-------------------


**Post merge**
